src/librustc_mir/transform/inline.rs

```diff
@@ -198,6 +198,16 @@ impl<'a, 'tcx> Inliner<'a, 'tcx> {
         }
     }
 
+    fn consider_optimizing(&self,
+                           callsite: CallSite<'tcx>,
+                           callee_mir: &Mir<'tcx>)
+                           -> bool
+    {
+        debug!("consider_optimizing({:?})", callsite);
+        self.tcx.consider_optimizing(|| format!("Inline {:?} into {:?}", callee_mir.span, callsite))
+            && self.should_inline(callsite, callee_mir)
```

Comment:
I think the order here is wrong, IMO we should first check if we can inline, that would reduce a lot of fuel uses and make it easier to track down a specific misinline.

Reply:
that makes sense to me

Reply:
Sure, I can do.